react-components/src/components/NodeCacheProvider/PointCloudAnnotationCache.ts

```diff
@@ -146,8 +160,10 @@ export class PointCloudAnnotationCache {
       revisionId
     );
 
-    return fetchedAnnotationAssetMappings.find((assetMap) =>
-      Array.from(assetMap.values()).some((asset) => asset.id === assetId)
+    const assetIdNumber = typeof assetId === 'number' ? assetId : undefined;
+    const matchedAnnotations = Array.from(fetchedAnnotationAssetMappings.entries()).find(
+      ([, asset]) => asset.id === assetIdNumber
     );
+    return matchedAnnotations !== undefined ? new Map([matchedAnnotations]) : undefined;
```

Comment:
Are we just returning a map with a single value in it? Can't we just return the list of annotation(s) instead?